math/uint.go

```diff
@@ -103,7 +103,12 @@ func (u Uint) Add(u2 Uint) Uint { return NewUintFromBigInt(new(big.Int).Add(u.i,
 func (u Uint) AddUint64(u2 uint64) Uint { return u.Add(NewUint(u2)) }
 
 // Sub adds Uint from another
-func (u Uint) Sub(u2 Uint) Uint { return NewUintFromBigInt(new(big.Int).Sub(u.i, u2.i)) }
+func (u Uint) Sub(u2 Uint) Uint {
+	if u.LT(u2) {
```

Comment:
This would be a reasonable step but the [UintOverflow](https://github.com/cosmos/cosmos-sdk/blob/v0.50.6/math/uint.go#L214) is ensuring the result is not negative and has the right bit length.